ucsmsdk/utils/comparesyncmo.py

```diff
@@ -0,0 +1,489 @@
+                              display properties which are equal
+        no_version_filter (bool): by default False: If set to True, ignore
+                                  properties which is introduced in later ucsm
+                                  version than reference ucsm
+        include_operational (bool): by default False: If set to True, compares
+                                    all the properties of mo.
+        xlate_org (str): org-dn, should be used compare objects of same type
+                         under different org. org-dn of reference mo.
+        xlate_map (dict) : {"difference_dn": "reference_dn"}
+                           Should be used to compare objects of same type with
+                           different dn.
+    Returns:
+        List of MoDiff objects:
+        MoDiff Object : dn - dn of Managed Object
+                        input_object - Managed Object
+                        side_indicator -
+                            "=>" Add the diff obj to ref ucsm
+                              or
+                              Modify the ref object by diff object on ref ucsm
+
+                            "<=" Removes the ref obj from ref ucsm
+
+                            "==" object is equal on both ref and diff ucsm
+                        diff_property  - property list with different value
+
+    Example:
+        #1
+        ref_mos = [ref_handle.query_dn(dn="org-root/ls-sp")]
+        diff_mos = [diff_handle.query_dn(dn="org-root/ls-sp")]
+        compare_ucs_mo(ref_handle, ref_mos,
+                                   diff_handle, diff_mos)
+
+        #2
+        ref_mos = [ref_handle.query_dn(dn="org-root/org-ref/ls-sp")]
+        diff_mos = [diff_handle.query_dn(dn="org-root/org-diff/ls-sp")]
+        compare_ucs_mo(ref_handle, ref_mos,
+                                   diff_handle, diff_mos,
+                                   xlate_org = "org-root/org-ref")
+
+        #3
+        ref_mos = [ref_handle.query_dn(dn="org-root/ls-ref")]
+        diff_mos = [diff_handle.query_dn(dn="org-root/ls-diff")]
+        compare_ucs_mo(ref_handle, ref_mos, diff_handle, diff_mos,
+                        xlate_map = {"org-root/ls-diff": "org-root/ls-ref"})
+    """
+
+    reference_dict = {}
+    difference_dict = {}
+
+    if ref_obj is not None and \
+            isinstance(ref_obj, list) and len(ref_obj) > 0:
+        for mo in ref_obj:
+            if mo is None:
+                continue
+            if isinstance(mo, GenericMo):
+                if ucsgenutils.word_u(mo.get_class_id()) in skip_mos:
+                    continue
+            elif mo.get_class_id() in skip_mos or \
+                    mo.mo_meta.inp_out == MoMeta.ACCESS_TYPE_OUTPUTONLY or \
+                    (mo.mo_meta.inp_out == MoMeta.ACCESS_TYPE_IO
+                     and len(mo.mo_meta.access) == 1
+                     and mo.mo_meta.access[0] == "read-only"):
+                continue
```

Comment:
@ragupta-git the above 10 lines are also used more than once. we should take out this piece of code to a method. DRY